cpp/order/RotationalAutocorrelation.cc

```diff
@@ -70,7 +69,7 @@ void RotationalAutocorrelation::compute(const quat<float>* ref_orientations, con
         for (unsigned int b = 0; b <= m_l; b++)
         {
             unit_harmonics.push_back(std::conj(hypersphere_harmonic(xi, zeta, a, b)));
-            prefactors[a][b] = m_factorials[a] * m_factorials[m_l - a] * m_factorials[b]
+            prefactors[a][b] = static_cast<float>(m_factorials[a]) * m_factorials[m_l - a] * m_factorials[b]
                 * m_factorials[m_l - b] / (float(m_l) + 1);
```

Comment:
This seems undesirable to cast only one factorial value. Make that consistent (either cast every value or cast their product). Also use `static_cast<float>(m_l)`.
```suggestion
            prefactors[a][b] = static_cast<float>(m_factorials[a]) * m_factorials[m_l - a] * m_factorials[b]
                * m_factorials[m_l - b] / (static_cast<float>(m_l) + 1);
```

Reply:
The intent was to cast the product so that the multiplication is integer multiplication, looks like I misplaced the closing parenthesis.